_posts/2019-09-20-lambda-to-openfaas.md

```diff
@@ -0,0 +1,314 @@
+- Replaced the AWS specific response object with the template's `context`
+- Set the `context.success` and `context.fail` status codes and values
+
+If you look at the `git diff --stat` output, we actually removed more code than we added!
+
+```shell
+ 1 file changed, 35 insertions(+), 37 deletions(-)
+```
+
+Also, notice that the AWS credentials are not in the code directly, so this function is still safe to commit to any repository without fear of leaking your credentials.
+
+> Because of the changes made to configuring the aws-sdk and reading secrets, this migrated code will no longer run on Lambda
+
+#### Build and Deploy
+Now that our function code is migrated, it's time to build and deploy it!
+
+Both of these can be accomplished with the `faas-cli` tool. Optionally, with [OpenFaaS Cloud](https://docs.openfaas.com/openfaas-cloud/), this can all happen automatically when the changes are pushed to Github or GitLab.
+
+- Ensure your CLI is logged in to your cluster
+```shell
+$ echo $PASSWORD | faas-cli login --password-stdin
+```
+
+- Now, we're ready to build and deploy the container images.
+```shell
+$ faas-cli up
+```
+
+> The `up` command combines the `build`, `push`, and `deploy` commands into one cohesive operation.
+
+
+#### Invoking the Function
+The functions are invoked via HTTP through the gateway
+```shell
+$ curl -d '{"url": "https://something.com/"}' http://127.0.0.1:8080/function/shortener
+```
+
+You can also invoke a function using the `faas-cli`
+```shell
+$ echo '{ "url": "https://something.com/" }' | faas-cli invoke shortener
+```
+
+
+## Wrapping Up
+We've migrated our function code from being restricted to only deployable on AWS Lambda infrastructure to being completely portable by using OpenFaaS. 
+
+Additionally, this means that developers are able to test their functions by deploying to a local Kubernetes cluster using Docker for Windows, Docker for Mac, [microk8s](https://johnmccabe.net/technology/projects/openfaas-on-microk8s/), or [KinD](https://blog.alexellis.io/be-kind-to-yourself/)
+
+To further expand the portability, we could migrate the database to MongoDB for a full OSS stack! An example of this can be found on [Alex's Repository](https://github.com/alexellis/mongodb-function)
+
+#### Connect With the Community
+
+  * [Join Slack now](https://goo.gl/forms/SqpLSdyzVoOboRqs1)
+  * [Contribute](https://docs.openfaas.com/contributing/get-started/)
+  * [Sponsor on GitHub](https://www.openfaas.com/support/)
+
+## For More Information 
+Learn the features of OpenFaaS in the [workshop](https://docs.openfaas.com/tutorials/workshop/)
+
+[Build your own OpenFaaS Cloud with AWS EKS](/blog/eks-openfaas-cloud-build-guide)
+
+[TLS & Custom Domains for your Functions](/blog/custom-domains-function-ingress/)
+
+mTLS, Traffic Splitting, and more with [OpenFaaS and Linkerd2](https://github.com/openfaas-incubator/openfaas-linkerd2)
```

Comment:
What's the link between Lambda and mTLS etc?

Reply:
There's no direct link. I added it as a link to show what other features are available now that the functions are on OpenFaaS. Something that's not possible with Lambda (as far as I know)